pandas/core/internals/blocks.py

```diff
@@ -928,10 +928,11 @@ def setitem(self, indexer, value):
 
             elif lib.is_scalar(value) and not isna(value):
                 dtype, _ = infer_dtype_from_scalar(value, pandas_dtype=True)
-
+            elif is_list_like(value) and is_empty_indexer(indexer, np.asarray(value)):
```

Comment:
im not sure this is the right place to catch this, since i think we are specifically interested in all-False boolean indexers coming through loc.  does this get here via setitem_with_indexer, and if so, can you point out specifically which branch(es)?

Reply:
Through ``_setitem_with_indexer_split_path``  -> ``_setitem_with_indexer_2d_value`` -> ``_setitem_single_column``

as mentioned in https://github.com/pandas-dev/pandas/pull/37672#discussion_r523415484 we could do that in ``setitem_with_indexer`` if this is better

Reply:
In _setitem_with_indexer_split_path L 1686-1688 we have a no-op block that is for all-false boolean masks.  could move it up to before the 2D check.  would that do the trick?

Reply:
One possible solution is moving up the ``lplane_indexer==0`` condition and removing the value part. Probably the shortest and does not seem to break tests. Don't know if this kills behavior not tested, but if indexer has len=0, we probably do not want to set values anyway